packages/api-rx/src/index.ts

```diff
@@ -0,0 +1,297 @@
+// Copyright 2017-2018 @polkadot/api-rx authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { RxApiInterface, QueryableStorageFunction, QueryableModuleStorage, QueryableStorage, SubmittableExtrinsics, SubmittableModuleExtrinsics, SubmittableExtrinsicFunction } from './types';
+
+import E3 from 'eventemitter3';
+import { EMPTY, Observable, from } from 'rxjs';
+import { defaultIfEmpty, map } from 'rxjs/operators';
+import WsProvider from '@polkadot/rpc-provider/ws';
+import RpcRx from '@polkadot/rpc-rx/index';
+import { Extrinsics, ExtrinsicFunction } from '@polkadot/extrinsics/types';
+import extrinsicsFromMeta from '@polkadot/extrinsics/fromMetadata';
+import { Storage } from '@polkadot/storage/types';
+import storageFromMeta from '@polkadot/storage/fromMetadata';
+import { Base } from '@polkadot/types/codec';
+import { Hash, Method, RuntimeVersion } from '@polkadot/types/index';
+import RuntimeMetadata from '@polkadot/types/Metadata';
+import assert from '@polkadot/util/assert';
+import isUndefined from '@polkadot/util/is/undefined';
+import logger from '@polkadot/util/logger';
+
+import SubmittableExtrinsic from './SubmittableExtrinsic';
+import { StorageFunction } from '@polkadot/types/StorageKey';
+
+const l = logger('api-rx');
+
+const INIT_ERROR = `Api needs to be initialised before using, listen on 'whenReady'`;
+
+export default class ApiRx extends E3.EventEmitter implements RxApiInterface {
+  private _extrinsics?: SubmittableExtrinsics;
+  private _genesisHash?: Hash;
+  private _storage?: QueryableStorage;
+  private _rpc: RpcRx;
+  private _runtimeMetadata?: RuntimeMetadata;
+  private _runtimeVersion?: RuntimeVersion;
+
+  // Observable that returns the first time we are connected and loaded
+  readonly isReady: Observable<ApiRx>;
+
+  /**
+   * @description Creates an ApiRx instance using the supplied provider. Returns an Observable containing the actual Api instance.
+   * @param wsProvider Optional WebSocket provider that is passed to the class contructor
+   * @example
+   * <BR>
+   *
+   * ```javascript
+   * import Api from '@polkadot/api-rx';
+   *
+   * Api.create().subscribe((api) => {
```

Comment:
should we use `ApiRx` instead of `Api` on lines 48 and 50?

Reply:
O prefer the shorter version - having said that since we are also exporting `ApiRx` now on index, it would depend on the import. Will have to re-visit when promise is there in a day or so.